src/core/libmaven/isotopeDetection.cpp

```diff
@@ -432,10 +432,10 @@ void IsotopeDetection::addIsotopes(PeakGroup* parentgroup,
         PeakGroup& child = (*itr2).second;
         child.metaGroupId = parentgroup->metaGroupId;
 
-        bool isotopeAdded = addIsotopes(parentgroup, child, isotopeName, C13Flag, N15Flag, S34Flag, D2Flag);
+        childStatistics(parentgroup, child, isotopeName);
+        bool isotopeAdded = filterLabel(isotopeName, C13Flag, N15Flag, S34Flag, D2Flag);
```

Comment:
filterLabel returns true if the isotope should not be added to parent. Maybe change 'isotopeAdded' to 'filterIsotope'? and continue if filterIsotope is true.